flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1191,7 +1191,7 @@ private TaskExecutorJobServices registerNewJobAndCreateServices(
         permanentBlobService.registerJob(jobId);
 
         return TaskExecutorJobServices.create(
-                libraryCacheManager.registerClassLoaderLease(jobId),
+                libraryCacheManager.registerClassLoaderLease(jobId, false),
```

Comment:
Good point! Currently we can use system classloader in TaskManager directly even when the `classpaths` is empty. That's because the index number for the name of generated class in flink sql is per job, and there will be the same class name in different job. If we use system classloader for sql in TaskManager, it will cause confliction for these classes. We have created an issue for TaskManager: https://issues.apache.org/jira/browse/FLINK-28691